src/txgraph.cpp

```diff
@@ -289,6 +290,8 @@ class TxGraphImpl final : public TxGraph
 
     /** Index of ChunkData objects. */
     ChunkIndex m_chunkindex;
+    /** Number of index-observing objects in existence (BlockBuilderImpl). */
+    size_t m_chunkindex_observers{0};
```

Comment:
Perhaps this stuff could be a little bit more RAII-ish? Something like:

```c++
class RefsVector : public std::vector<Ref*>
{
    TxGraph& m_graph;
public:
    template<typename... T>
    RefsView(TxGraph& graph, T&&... args) : std::vector<Ref*>{args...}, m_graph{graph} { ++m_graph.m_chunkindex_observers; }
    ~RefsView() { --m_graph.m_chunkindex_observers; }
};
```

Might be good to introduce `m_chunkindex_observers` in its own commit? Arguably having a vector of Ref's (making it invalid to do `Ref b = std::move(a);`) is a different constraint from being unable to change the linearisation at the main level (which would mess up BlockBuilder); could potentially have distinct "observer" values for each, which would allow fees to be bumped, when you just have a Ref vec, eg.

Reply:
> It's a possibility, but you mean this in addition to `BlockBuildImpl` itself also incrementing the observers count, right? Because it holds an iterator to the chunk index in GraphImpl too.

Yes. (though, technically, if `BlockBuildImpl` stores its own RefsVector it could avoid explicitly incrementing the observers count, because RefsVector does that for it)